### PR TITLE
Add build script for vim

### LINKED ISSUE #124: Build script for vim (only text-mode is needed)
Need a build script for a new version of vim.

### PR DESCRIPTION
**Pull request by Alan-Kuan**
I added a build script for vim with basic functionality.
Since there is no release in tarball format with its md5sum for vim, I only added the "git" source.
Please take a look at my PR. Thanks.

This resolves #124.

### CODE REVIEW

File: scripts/build.d/vim [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -e
+
+SRCSYNC=${SRCSYNC:-tarball}
```
Comment (yungyuc) [inline scripts/build.d/vim:5]: You are saying to use a tarball by default here.

File: scripts/build.d/vim [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+pkgname=vim
+pkgbranch=${VERSION:-master}
+pkgfull=$pkgname
+
+# unpack (clone)
+syncgit https://github.com/${pkgname} ${pkgname} ${pkgbranch} ${pkgfull}
```
Comment (yungyuc) [inline scripts/build.d/vim:12]: However, here you choose to unconditionally get the source by using git. It is inconsistent to what you write in line # 5.

Please use a tarball by default.  Pulling from git is more time-consuming, less stable, and should serve as a secondary approach.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Please fix the issue of inconsistency and improve the stability.

File: scripts/build.d/vim (hunk 1 above)
Comment (Alan-Kuan) [reply scripts/build.d/vim:5]: Oh. You're right. I just forgot it.

File: scripts/build.d/vim (hunk 2 above)
Comment (Alan-Kuan) [reply scripts/build.d/vim:12]: @yungyuc
I see. However, the official repo of vim doesn't provide md5sum of a specific tarball file.
Should I download it and compute by myself (which I think is not a good practice)?
Or perhaps there's actually a md5sum coming along with a specific tarball file that I just did not know?

image

Comment (Alan-Kuan) [reply scripts/build.d/vim:12]: I got it! I will update the script soon.

**Comment by Alan-Kuan** [comment]
I finished the tarball support.
btw, the testing process took me so long that I ended up terminating it.
Thus, I only made sure the build script could compile and install vim successfully.

**Review by yungyuc — APPROVED** [review]

**Comment by yungyuc** [comment]
LGTM.  Let's get it merged anyway.

What do you mean by "took me so long that I ended up terminating it" while "made sure the build script could compile and install vim successfully"?  They seem to contradict to each other.

**Comment by Alan-Kuan** [comment]
@yungyuc
Oh. I just commented out the line for testing it and re-run the build script.
After it finished, I manually test the program and uncommented that line.

**Comment by yungyuc** [comment]
For how long it took so that you need to comment out that line.  And which line you are referring to?

**Comment by yungyuc** [comment]
I think you meant https://github.com/solvcon/devenv/blob/d17735c514c01c3813994a917d9e5492e2623ac6/scripts/build.d/vim#L50 ?

I overlooked it.  It should not kick in automatically because it does not make sense to run tests with installation.  It is not a bad thing to have it in the build script, but it should be made opt-in.

Could you use a new PR to turn it off?  Maybe use a environment variable so that it can be turned on without changing the script.

**Comment by Alan-Kuan** [comment]
Sure. I will fix it.

**Pull request merged**